pkg/datapath/linux/bigtcp/bigtcp.go

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package bigtcp
+
+import (
+	datapathOption "github.com/cilium/cilium/pkg/datapath/option"
+
+	"github.com/vishvananda/netlink"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/probes"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+const (
+	defaultGROMaxSize int = 65536
+	defaultGSOMaxSize int = 65536
+
+	BIGTCPGROMaxSize int = 196608
+	BIGTCPGSOMaxSize int = 196608
+)
+
+// Configuration is a BIG TCP configuration as returned by NewConfiguration
+type Configuration struct {
+	// gsoMaxSize is the GSO maximum size used when configuring devices
+	//
+	// Note that this is a singleton for the process including this
+	// package. This means, for instance, that when using this from the
+	// ``pkg/plugins/*`` sources, it will not respect the settings
+	// configured inside the ``daemon/``.
+	gsoMaxSize int
+
+	// groMaxSize is the GRO maximum size used when configuring devices
+	//
+	// Note that this is a singleton for the process including this
+	// package. This means, for instance, that when using this from the
+	// ``pkg/plugins/*`` sources, it will not respect the settings
+	// configured inside the ``daemon/``.
+	groMaxSize int
+}
+
+func (c *Configuration) GetGROMaxSize() int {
+	return c.groMaxSize
+}
+
+func (c *Configuration) GetGSOMaxSize() int {
+	return c.gsoMaxSize
+}
+
+func setGROGSOMaxSize(device string, GROMaxSize, GSOMaxSize int) error {
+	link, err := netlink.LinkByName(device)
+	if err != nil {
+		log.WithError(err).WithField("device", device).Warn("Link does not exist")
+		return nil
+	}
+
+	attrs := link.Attrs()
+	if (int(attrs.GROMaxSize) == GROMaxSize && int(attrs.GSOMaxSize) == GSOMaxSize) ||
+		(!option.Config.EnableIPv6BIGTCP &&
+			int(attrs.GROMaxSize) <= GROMaxSize &&
+			int(attrs.GSOMaxSize) <= GSOMaxSize) {
+		return nil
```

Comment:
The second case here, 
```
!EnableIPv6BIGTCP && attrs.GROMaxSize <= GROMaxSize && GSOMaxSize <= GSOMaxSize
```
this is an error right? The user tried to set GRO/GSO larger than the max should we return an error?

Reply:
no, that's to avoid changing gso/gro when not necessary, we have the default case where big tcp is disabled and the case where it was enabled and then disabled, this is used to match the first one

Reply:
I'll add a comment